packages/talos_agent/src/messaging/kafka.rs

```diff
@@ -137,49 +135,47 @@ impl KafkaConsumer {
     }
 
     pub fn subscribe(&self) -> Result<(), String> {
-        let result = match self.consumer.subscribe(&[self.config.certification_topic.as_str()]) {
-            Ok(_) => {
-                loop {
-                    match self.consumer.fetch_group_list(None, Duration::from_secs(1)) {
-                        Ok(group_list) => {
-                            let list = group_list.groups();
-                            let mut found = false;
-                            for group_info in list {
-                                if group_info.name() == self.config.group_id {
-                                    log::info!("Detected consumer group: name: {}, state: {}", group_info.name(), group_info.state());
-
-                                    if group_info.state() != "Stable" {
-                                        continue;
-                                    }
-
-                                    found = true;
-                                    break;
-                                }
-                            }
-
-                            if found {
-                                break;
-                            }
-                        }
-                        Err(_) => {
-                            log::error!("Cannot fetch group info for group id '{}'", self.config.group_id);
-                            break;
+        let topic = self.config.certification_topic.as_str();
+        let partition = 0_i32;
+
+        let mut partition_list = TopicPartitionList::new();
+        partition_list.add_partition(topic, partition);
+        // This line is required for seek operation to be successful.
+        partition_list.set_partition_offset(topic, partition, Offset::End).unwrap();
+
+        let rslt_assign = self.consumer.assign(&partition_list);
+        match rslt_assign {
+            Ok(()) => {
+                return match self.get_offset(partition, Duration::from_secs(5)) {
+                    Ok(offset) => {
+                        log::info!("Seeking on partition {} to offset: {:?}", partition, offset);
+                        match self.consumer.seek(topic, partition, offset, Duration::from_secs(5)) {
```

Comment:
Too many nested `match` statements can make it hard to read. 

And example to simplify this would be:-

```rust
self.consumer
                            .seek(topic, partition, offset, Duration::from_secs(5))
                            .map_err(|e| format!("Error when Seeking to offset {:?}. Error: {}", offset, e))
```

Reply:
Flattening the entire code would be as follows:- 

```rust
rslt_assign.map_err(|e| format!("Error invoking assign('{}', {}. Error: {})", topic, partition, e))?;

let offset = self
    .get_offset(partition, Duration::from_secs(5))
    .map_err(|e| format!("Error fetching offset of partition {}. Error: {})", partition, e))?;

Ok(self
    .consumer
    .seek(topic, partition, offset, Duration::from_secs(5))
    .map_err(|e| format!("Error when Seeking to offset {:?}. Error: {}", offset, e))?)
```

Reply:
No nested match statements. But I was able to do this because of the  `?` operator. You can read more about it here https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html

Reply:
I like it better. Thanks!

Reply:
Done